components/automate-ui/src/app/modules/user/user-details/user-details.component.ts

```diff
@@ -34,99 +41,183 @@ export type UserTabName = 'password' | 'details';
+
+  constructor() {}
+
+  abstract savePassword(): void;
+  abstract saveDisplayName(): void;
+  protected abstract createPasswordForm(fb: FormBuilder): void;
+
+  protected resetForms(): void {
+    this.displayNameForm.reset();
+    if (this.user) {
+      this.displayNameForm.patchValue({displayName: this.user.name});
+    }
+    this.passwordForm.reset();
+  }
+
+  protected createForms(fb: FormBuilder): void {
     // Must stay in sync with error checks in user-details.component.html
     this.displayNameForm = fb.group({
       displayName: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
     });
+
+    this.createPasswordForm(fb);
+  }
+}
+
+// This view is used when a user goes to another users detail page.
+class UserAdminDetails extends UserDetails {
+  public loading$: Observable<boolean>;
+  public user: User;
+  public saveSuccessful = false;
+  public saveInProgress = false;
+  public displayNameForm: FormGroup;
+  public passwordForm: FormGroup;
+  public showBreadcrumbs = true;
+  public showPreviousPassword = false;
+
+  constructor(
+    userId: string,
+    private store: Store<NgrxStateAtom>,
+    layoutFacade: LayoutFacadeService,
+    isDestroyed: Subject<boolean>,
+    fb: FormBuilder) {
+      super();
+
+      this.store.dispatch(new GetUser({ id: userId }));
+
+      layoutFacade.showSettingsSidebar();
+      this.createForms(fb);
+      this.resetForms();
+
+      this.loading$ = this.store.select(getUserStatus).pipe(
+        map((status: EntityStatus) =>  status !== EntityStatus.loadingSuccess));
+
+      combineLatest([
+        this.loading$,
+        this.store.select(userFromRoute)
+      ]).pipe(
+        filter(([loadingUserData, user]) => !loadingUserData && !isNil(user)),
+        map(([_, user]) => user),
+        takeUntil(isDestroyed)
+      ).subscribe((user) => {
+          this.user = { ...user };
+          this.displayNameForm.patchValue({displayName: this.user.name});
+      });
+
+      this.store.select(updateUserStatus).pipe(
+        filter(status => this.saveInProgress && !loading(status)),
+        takeUntil(isDestroyed)
+      ).subscribe((status) => {
+          this.saveInProgress = false;
+          // same status is used for updating password or display name, so we just reset both
+          this.saveSuccessful = (status === EntityStatus.loadingSuccess);
+          if (this.saveSuccessful) {
+            this.resetForms();
+          }
+      });
+  }
+
+  public savePassword(): void {
+    this.saveSuccessful = false;
+    this.saveInProgress = true;
+    const password = this.passwordForm.get('newPassword').value;
+    this.store.dispatch(new UpdatePasswordUser({ ...this.user, password }));
+  }
+
+  public saveDisplayName(): void {
+    this.saveSuccessful = false;
+    this.saveInProgress = true;
+    const name = this.displayNameForm.get('displayName').value.trim();
+    this.store.dispatch(new UpdateNameUser({ ...this.user, name }));
```

Comment:
Last tidbit about streamlining abstract classes: by defining the method first in the base class...
```
 protected saveDisplayName(): void {
    this.saveSuccessful = false;
    this.saveInProgress = true;
    const name = this.displayNameForm.get('displayName').value.trim();
 }
```
Then this method in the derived class becomes this--which very clearly shows what is unique about it. 
```
public saveDisplayName(): void {
  super.saveDisplayName();
  this.store.dispatch(new UpdateNameUser({ ...this.user, name }));
}
```
Just a suggestion.

Reply:
I think this makes what is going on more confusing because I would have to look between the two functions to really see what is going on. 

I could see doing the below though. 
```
// base class
 public saveDisplayName(): void {
    this.saveSuccessful = false;
    this.saveInProgress = true;
    const name = this.displayNameForm.get('displayName').value.trim();
    this.store.dispatch(this.createUpdateUserNameAction( name ));
 }

abstract createUpdateUserNameAction(name: string): Action;
```
```
// derived class. 
createUpdateUserNameAction(name: string): Action {
   return new UpdateNameUser({ ...this.user, name });
}
```

Reply:
Agreed--I like that better! 👍